### ISSUE TITLE
Build failure: arrow-cpp (staging branch)

### ISSUE THREAD

**Issue by superherointj** [NixOS/nixpkgs#322003]
### Steps To Reproduce

At **staging** (not `master`):
1. nix build .#arrow-cpp

### Build log

```
--   Outputting build configuration summary to /build/apache-arrow-16.0.0/cpp/build/cmake_summary.json
-- Configuring done (13.8s)
CMake Error at src/arrow/CMakeLists.txt:592 (target_link_libraries):
  Target "arrow_util" links to:

    LZ4::lz4

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```
Full log: https://termbin.com/i0tg

### Additional context

(I've opened this to log or look after a solution for this.)

* For debug purposes, reverting #318512 fixes `arrow-cpp` build.

### Notify maintainers

CC @tobim @veprbl @cpcloud

**Comment by tobim** [comment]
Most likely the caused by #318512 ([comment](https://github.com/NixOS/nixpkgs/pull/318512#issuecomment-2182851656)).

**Comment by superherointj** [comment]
Fixed at #322021
